frontend/appflowy_flutter/lib/plugins/document/application/template_service.dart

```diff
@@ -0,0 +1,246 @@
+
+    final dbFile = File('${directory.path}/template/$name.csv');
+    await dbFile.writeAsString(pb);
+  }
+
+  /// Steps:
+  /// 1. Pick template(.zip)
+  /// 2. Zip may contain several files, use [config.json] to determine which files to use.
+  /// 3. Load template into editor, using [TemplateService.unloadTemplate] function
+
+  Future<Archive?> pickTemplate() async {
+    // Pick a ZIP file from the system
+    final result = await FilePicker().pickFiles(
+      type: FileType.custom,
+      allowedExtensions: ['zip'],
+      allowMultiple: false,
+    );
+
+    // User cancelled the picker
+    if (result == null) return null;
+
+    // Extract the contents of the ZIP file
+    final file = File(result.files.single.path!);
+
+    final contents = await file.readAsBytes();
+    final archive = ZipDecoder().decodeBytes(contents);
+
+    return archive;
+  }
+
+  Future<void> unloadTemplate(
+    String parentViewId,
+    Archive? archive,
+  ) async {
+    if (archive == null) return;
+
+    final directory = await getTemporaryDirectory();
+
+    for (final file in archive) {
+      final filename = '${directory.path}/${file.name}';
+      final data = file.content as List<int>;
+      final outputFile = File(filename);
+      await outputFile.create(recursive: true);
+      await outputFile.writeAsBytes(data);
+    }
+
+    final config =
+        json.decode(await File("${directory.path}/config.json").readAsString());
+
+    final TemplateModel template = TemplateModel.fromJson(config);
+
+    debugPrint("Loading Template:  ${template.templateName} into editor");
+
+    for (final DocumentModel doc in template.documents) {
+      _loadTemplate(parentViewId, doc);
+    }
+  }
+
+  Future<void> _loadTemplate(String parentViewId, DocumentModel doc) async {
+    final directory = await getTemporaryDirectory();
+
+    /// Import all databases first
+
+    final List<ViewPB> dbViews = [];
+
+    for (final db in doc.db) {
+      final res = await _importDB(db, parentViewId);
+      if (res == null) {
+        // Abort if an error occurred while importing the database
+        debugPrint("TemplateService: An error occurred while importing $db");
+        return;
+      }
+
+      dbViews.add(res);
+    }
+
+    /// Import the document and embed the [dbViews]
+
+    final String templateRes =
+        await File('${directory.path}/${doc.name}').readAsString();
+
+    final Map<String, dynamic> docJson = json.decode(templateRes);
+
+    final List<dynamic> children = docJson["document"]["children"];
+
+    int dbCounter = 0;
+
+    if (dbViews.isNotEmpty) {
+      for (final child in children) {
+        if (child["type"] == "grid" || child["type"] == "board") {
+          /// Update old ID's with new ID's
+          child["data"]["view_id"] = dbViews[dbCounter++].id;
+          child["data"]["parent_id"] = parentViewId;
+        }
+      }
+    }
+
+    final document = Document.fromJson({
+      "document": {
+        "type": "page",
```

Comment:
same here. use the const value.

Reply:
"page" is not defined in the `DatabaseBlockKeys`. Should I define it above this?